predicators/nsrt_learning/strips_learning/effect_search_learner.py

```diff
@@ -202,79 +174,55 @@ def _get_first_uncovered_transition(
-                        # (1) the preconditions hold in the pre-image state of
-                        # the segment.
-                        # (2) the effects yield a state that is a subset of the
-                        # post-image state.
-                        # (3) the PNAD's add effects already capture all the
-                        # necessary add effects.
-                        # This means that the only reason this PNAD failed to
-                        # capture the necessary image is due to some issue
-                        # with delete/ignore effects.
-                        pred_next_atoms = utils.apply_operator(
-                            last_used_op, atoms_seq[t])
-                        missing_effects = (necessary_image - pred_next_atoms)
-                        # These are just the missing effects + existing add
-                        # effects.
-                        necessary_add_effects = missing_effects | \
-                            last_used_op.add_effects
-                        # These are the missing effects that were ignore
-                        # effects of the last_used_op.
-                        necessary_keep_effects = {
-                            a
-                            for a in missing_effects if a in atoms_seq[t] and (
-                                a.predicate in last_used_op.ignore_effects
-                                or a in last_used_op.delete_effects)
-                        }
-
-                    return (option.parent, option.objects,
-                            necessary_add_effects, necessary_keep_effects)
+                    return segment
         return None
 
 
 class _PruningEffectSearchOperator(_EffectSearchOperator):
     """An operator that prunes effect sets."""
 
-    def get_successors(self,
-                       effect_sets: _EffectSets) -> Iterator[_EffectSets]:
-        for (spec, add_effects, keep_effects) in effect_sets:
-            yield effect_sets.remove(spec, add_effects, keep_effects)
+    def get_successors(
+        self, pnads: FrozenSet[PartialNSRTAndDatastore]
+    ) -> Iterator[FrozenSet[PartialNSRTAndDatastore]]:
+        # NOTE: Sorting done here to maintain determinism.
+        for pnad_to_remove in sorted(pnads,
+                                     key=lambda p: len(p.op.add_effects)):
+            yield frozenset([pnad for pnad in pnads if pnad != pnad_to_remove])
 
 
 class _EffectSearchHeuristic(abc.ABC):
     """Given a set of effect sets, produce a score, with lower better."""
 
     def __init__(
-        self,
-        trajectories: List[LowLevelTrajectory],
-        train_tasks: List[Task],
-        predicates: Set[Predicate],
-        segmented_trajs: List[List[Segment]],
-        effect_sets_to_pnads: Callable[[_EffectSets],
-                                       List[PartialNSRTAndDatastore]],
+        self, trajectories: List[LowLevelTrajectory], train_tasks: List[Task],
```

Comment:
same comment about passing in the class instance itself rather than these methods